app/controllers/congress_messages_controller.rb

```diff
@@ -0,0 +1,101 @@
+class CongressMessagesController < ApplicationController
+  include Tooling
+
+  before_action :set_congress_message_campaign
+  before_action :update_user, only: :create
+  before_action :subscribe_user, only: :create
+
+  rescue_from SmartyStreets::AddressNotFound, with: :address_not_found
+  rescue_from CongressForms::RequestFailed, with: :congress_forms_request_failed
+
+  def new
+    if @campaign.target_bioguide_ids.present?
+      bioguide_ids = @campaign.target_bioguide_ids.split
+    else
+      location = SmartyStreets.get_location(params["street_address"], params["zipcode"])
+      members = @campaign.targets.for_district(location.state, location.district)
+      bioguide_ids = members.pluck(:bioguide_id)
+    end
+
+    forms = CongressForms::Form.find(bioguide_ids)
+    @message = CongressMessage.new_from_lookup(location, params[:message], @campaign, forms)
+    render partial: "form"
+  end
+
+  def create
+    @message = CongressMessage.new(congress_message_params.merge(campaign: @campaign))
+    @message.forms = CongressForms::Form.find(params["bioguide_ids"].split)
+
+    if @message.submit
+      @name = user_params[:first_name] # for deliver_thanks_message
```

Comment:
I'd like to start moving some of the helpers in `tools_controller.rb` to the tooling concern. Many of them rely really heavily on instance variable being set, which I don't think is ideal, but I didn't want to move too much more around for this PR.